vsphere/data_source_vsphere_host_thumbprint.go

```diff
@@ -0,0 +1,49 @@
+package vsphere
+
+import (
+	"bytes"
+	"crypto/sha1"
+	"crypto/tls"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceVSphereHostThumbprint() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceVSphereHostThumbprintRead,
+		Schema: map[string]*schema.Schema{
+			"address": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The address of the ESXi to extract the thumbprint from.",
+			},
+			"port": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "443",
+				Description: "The port to connect to on the ESXi host.",
+			},
+		},
+	}
+}
+
+func dataSourceVSphereHostThumbprintRead(d *schema.ResourceData, meta interface{}) error {
+	config := &tls.Config{}
+	config.InsecureSkipVerify = true
+	conn, err := tls.Dial("tcp", d.Get("address").(string)+":"+d.Get("port").(string), config)
+	if err != nil {
+		return err
+	}
+	cert := conn.ConnectionState().PeerCertificates[0]
```

Comment:
We should mention explicitly in the documentation that if the host is using a cert chain then we will only return the fingerprint of the first cert we find.

Reply:
Good call. Added.